src/FakeItEasy/Core/FakeWrapperConfigurator.cs

```diff
@@ -1,30 +1,18 @@
-namespace FakeItEasy.Core
+﻿namespace FakeItEasy.Core
 {
     using System;
     using System.Collections.Generic;
     using System.Linq.Expressions;
     using FakeItEasy.Creation;
-#if FEATURE_SELF_INITIALIZED_FAKES
-    using FakeItEasy.SelfInitializedFakes;
-#endif
 
     /// <summary>
     /// Handles configuring of fake objects to delegate all their calls to a wrapped instance.
     /// </summary>
     /// <typeparam name="T">The type of fake object generated.</typeparam>
     internal class FakeWrapperConfigurator<T>

```

Comment:
This class got quite thin all of a sudden. Consider removing it and making `DefaultFakeAndDummyManager` do all the work? Like so:

```diff
diff --git a/src/FakeItEasy/Creation/DefaultFakeAndDummyManager.cs b/src/FakeItEasy/Creation/DefaultFakeAndDummyManager.cs
index 73fca73..0a6ebec 100644
--- a/src/FakeItEasy/Creation/DefaultFakeAndDummyManager.cs
+++ b/src/FakeItEasy/Creation/DefaultFakeAndDummyManager.cs
@@ -100,9 +100,8 @@
            public override IFakeOptions<T> Wrapping(T wrappedInstance)
             {
                 Guard.AgainstNull(wrappedInstance, nameof(wrappedInstance));
 
-                var wrapper = new FakeWrapperConfigurator<T>(this, wrappedInstance);
-                this.ConfigureFake(fake => wrapper.ConfigureFakeToWrap(fake));
-                return wrapper;
+                this.ConfigureFake(fake => ConfigureFakeToWrap(fake, wrappedInstance));
+                return this;
             }
 
             public override IFakeOptions<T> Implements(Type interfaceType)
@@ -116,6 +115,11 @@
                 this.proxyOptions.AddProxyConfigurationAction(proxy => action((T)proxy));
                 return this;
             }

+            private static void ConfigureFakeToWrap(object fakedObject, object wrappedObject)
+            {
+                var fake = Fake.GetFakeManager(fakedObject);
+                fake.AddRuleFirst(new WrappedObjectRule(wrappedObject));
+            }
         }
     }
 }
```

(or similar)? In theory, then `FakeOptionsBase<T>` could be inlined after that, but maybe I'm being too aggressive.